src/transformers/modeling_tf_utils.py

```diff
@@ -1226,6 +1238,14 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
 
         config.name_or_path = pretrained_model_name_or_path
 
+        # composed models, *e.g.* TFRag, require special treatment when it comes to loading
+        # pre-trained weights.
+        if (
+            "load_weight_prefix" in inspect.signature(cls.__init__).parameters.keys()
```

Comment:
This test is black magic. Can we make a class attribute instead of looking if the init accepts `load_weight_prefix` as argument?

Reply:
Yeah we can change this a bit by introducing a class attribute `requires_load_weight_prefix` instead of using `inspect.signature(....)` which defaults to `False` and **only** for `TFRagModel` is set to `True`